src/kdbindings/connection_handle.h

```diff
@@ -177,10 +177,19 @@ class ConnectionHandle
         return false;
     }
 
+    // Factory method to create a ConnectionHandle
+    static std::shared_ptr<ConnectionHandle> create(const std::weak_ptr<Private::SignalImplBase>& signalImpl, std::optional<Private::GenerationalIndex> id) {
```

Comment:
Perhaps the create() function should be private? Signal is already a friend so that will still work. Or is there some other reason to expose this as public API?

Reply:
oops, no I think we can make it private. I was trying something inside
```
class Impl : public Private::SignalImplBase
``` 
and so made it public for that, but yeah fine for `Signal` class.  Thanks for pointing out!